File: backend/profile-service/src/main/java/com/example/profileservice/profile/repository/ProfileRepository.java
package com.example.profileservice.profile.repository;

import com.example.profileservice.profile.entity.Profile;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface ProfileRepository extends JpaRepository<Profile, Long>, ProfileCustomRepository {

    List<Profile> findByMemberId(Long memberId);
}

File: backend/profile-service/src/main/kotlin/com/example/profileservice/profile/repository/ProfileRepository.kt
package com.example.profileservice.profile.repository

import com.example.profileservice.profile.entity.Profile
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface ProfileRepository : JpaRepository<Profile, Long>, ProfileCustomRepository {

    fun findByMemberId(memberId: Long): List<Profile>
}
